packages/pyright-internal/src/analyzer/program.ts

```diff
@@ -1791,6 +1791,10 @@ export class Program {
             // Bind all of the implicit imports first. So we don't recurse into them.
             if (!isImplicitImport) {
                 this._bindImplicitImports(fileToBind);
+
+                if (!fileToBind.sourceFile.isBindingRequired()) {
```

Comment:
This solution looks good to me.

The need for this check isn't immediately obvious, however. Since there's no regression test for this, a comment explaining why it's needed is probably a good idea to prevent it from being accidentally modified or deleted in the future.